azure-kusto-data/azure/kusto/data/request.py

```diff
@@ -416,93 +424,56 @@ def _execute(self, endpoint, database, query, default_timeout, properties=None):
         :param str table: Target table.
         :param io.BaseIO stream: stream object which contains the data to ingest.
+        :param ClientRequestProperties client_request_properties: additional request properties.
         :param DataFormat stream_format: Format of the data in the stream.
         :param str mapping_name: Pre-defined mapping of the table. Required when stream_format is json/avro.
-        Other optional params: optional request headers as documented at:
-            https://kusto.azurewebsites.net/docs/api/rest/streaming-ingest.html
         """
 
         request_params = {"streamFormat": stream_format}
 
-        if stream_format in self._mapping_required_formats and mapping_name is not None:
+        if mapping_name is not None:
             request_params["mappingName"] = mapping_name
 
-        request_headers = {
-            "Accept-Encoding": accept_encoding,
-            "Connection": connection,
-            "x-ms-client-version": "Kusto.Python.StreamingClient:" + VERSION,
-            "x-ms-client-request-id": "KPSC.execute;" + str(uuid.uuid4()),
-            "Host": self._streaming_ingest_endpoint.split("/")[2],
-        }
-
-        if accept is not None:
-            request_headers["Accept"] = accept
-        if content_encoding is not None:
-            request_headers["Content-Encoding"] = content_encoding
-        if expect is not None:
-            request_headers["Expect"] = expect
-        if content_length is not None:
-            request_headers["Content-Length"] = str(content_length)
-        if self._auth_provider:
-            request_headers["Authorization"] = self._auth_provider.acquire_authorization_header()
-
-        response = self._session.post(
-            self._streaming_ingest_endpoint + database + "/" + table,
-            params=request_params,
-            headers=request_headers,
-            data=stream,
-            timeout=KustoClient._query_default_timeout,
-        )
+        request_headers = deepcopy(self._request_headers)
+        request_headers["x-ms-client-request-id"] = "KPC.execute_streaming_ingest;" + str(uuid.uuid4())
+        if client_request_properties:
+            request_headers.update(json.loads(client_request_properties.to_json())["Options"])
 
-        if response.status_code == 200:
-            return KustoResponseDataSetV1(response.json())
+        timeout = self._get_timeout(client_request_properties, self._streaming_ingest_default_timeout)
+        endpoint = self._streaming_ingest_endpoint + database + "/" + table
 
-        raise KustoServiceError([response.content], response)
+        self._post(endpoint, params=request_params, headers=request_headers, data=stream, timeout=timeout.seconds)
 
     def _get_timeout(self, properties, default):
         if properties:
             return properties.get_option(ClientRequestProperties.OptionServerTimeout, default)
         return default
 
+    def _post(self, url, **kwargs):

```

Comment:
The reason you took this method out, I guess, is that you have two _execute functions. Once you consolidate the two you can put this logic in the consolidation.